tests/functional/remote/03-symlink-dirs.t

```diff
@@ -0,0 +1,114 @@
+#-------------------------------------------------------------------------------
+# Checks configured symlinks are created for run, work, share, share/cycle, log 
+# # directories on localhost and the remote platform.
+. "$(dirname "$0")/test_header"
+require_remote_platform
+# shellcheck disable=SC2016
+
+if [[ -z ${TMPDIR:-} || -z ${USER:-} || $TMPDIR/$USER == "$HOME" ]]; then
+    skip_all '"TMPDIR" or "USER" not defined or "TMPDIR"/"USER" is "HOME"'
+fi
+
+set_test_number 12
+
+create_test_global_config "" "
+[symlink dirs]
+   [[localhost]]
+      run = \$TMPDIR/\$USER/cylctb_tmp_run_dir
+      share = \$TMPDIR/\$USER
+      log = \$TMPDIR/\$USER
+      share/cycle = \$TMPDIR/\$USER/cylctb_tmp_share_dir
+      work = \$TMPDIR/\$USER
+   [[$CYLC_TEST_PLATFORM]]
+      run = \$TMPDIR/\$USER/ctb_tmp_run_dir
+      share = \$TMPDIR/\$USER
+      log = \$TMPDIR/\$USER
+      share/cycle = \$TMPDIR/\$USER/ctb_tmp_share_dir
+      work = \$TMPDIR/\$USER
+"
+
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+
+run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}" \
+      -s "CYLC_TEST_PLATFORM=${CYLC_TEST_PLATFORM}"
+suite_run_ok "${TEST_NAME_BASE}-run-ok" cylc run "${SUITE_NAME}" \
+     -s "CYLC_TEST_PLATFORM=${CYLC_TEST_PLATFORM}"
+poll_grep_suite_log 'File installation complete.'
+TEST_SYM="${TEST_NAME_BASE}-run-symlink-exists-ok"
+if [[ $(readlink "$HOME/cylc-run/${SUITE_NAME}") == \
+"$TMPDIR/$USER/cylctb_tmp_run_dir/cylc-run/${SUITE_NAME}" ]]; then
+ok "$TEST_SYM.localhost"
+else
+    fail "$TEST_SYM.localhost"
+fi
+
+TEST_SYM="${TEST_NAME_BASE}-share-cycle-symlink-exists-ok"
+if [[ $(readlink "$HOME/cylc-run/${SUITE_NAME}/share/cycle") == \
+"$TMPDIR/$USER/cylctb_tmp_share_dir/cylc-run/${SUITE_NAME}/share/cycle" ]]; then
+ok "$TEST_SYM.localhost"
+else
+    fail "$TEST_SYM.localhost"
+fi
+
+for DIR in 'work' 'share' 'log'; do
+   TEST_SYM="${TEST_NAME_BASE}-${DIR}-symlink-exists-ok"
+   if [[ $(readlink "$HOME/cylc-run/${SUITE_NAME}/${DIR}") == \
+   "$TMPDIR/$USER/cylc-run/${SUITE_NAME}/${DIR}" ]]; then
+ok "$TEST_SYM.localhost"
+else
+    fail "$TEST_SYM.localhost"
+fi
+
+done
+
+SSH="$(cylc get-global-config -i "[platforms][$CYLC_TEST_PLATFORM]ssh command")"
+
+# shellcheck disable=SC2016
+
+LINK="$(${SSH} "${CYLC_TEST_HOST}" 'readlink "$HOME/cylc-run/'"$SUITE_NAME"'"')"
+if [[ "$LINK" == "$TMPDIR/$USER/ctb_tmp_run_dir/cylc-run/${SUITE_NAME}" ]]; then
+    ok "${TEST_NAME_BASE}-run-symlink-exists-ok.remotehost"
+else
+    fail "${TEST_NAME_BASE}-run-symlink-exists-ok.remotehost"
+fi
+# shellcheck disable=SC2016
+
+
+LINK="$(${SSH} "${CYLC_TEST_HOST}" 'readlink "$HOME/cylc-run/'"$SUITE_NAME"/share/cycle'"')"
+if [[ "$LINK" == "$TMPDIR/$USER/ctb_tmp_share_dir/cylc-run/${SUITE_NAME}/share/cycle" ]]; then
+    ok "${TEST_NAME_BASE}-share/cycle-symlink-exists-ok.remotehost"
+else
+    fail "${TEST_NAME_BASE}-share/cycle-symlink-exists-ok.remotehost"
+fi
+
+# shellcheck disable=SC2016
```

Comment:
Is this one needed? Perhaps it is, but on line 103?

Reply:
I think you are referring to the shellcheck disable? I have tried removing them and it complains if any of them are removed - I guess SSHing resets, unless there is a nifty way of globally setting it for the whole test.